packages/tectonic-explorer/src/stores/simulation-store.ts

```diff
@@ -210,6 +216,14 @@ export class SimulationStore {
     return result;
   }
 
+  @computed get arePinsLimited() {
+    return this.limitPins;
+  }
+
+  @computed get reachedPinLimit() {
+    return this.pinLimitReached;
+  }
```

Comment:
We don't really need this state variable - I think it's redundant. This computed property could check if `dataSamplesLength < 20`. That way you also limit the room for bugs, as this redundant state could get out of sync with the actual number of pins. Not a big deal here, but it's good practice to try to compute props as long as possible. One exception could be when calculations are too expensive and should be cached. Then, 
 this kind of redundant state becomes useful. But here it's only a simple comparison between two numbers.